### DISCUSSION TITLE
calls from -e are starting before register

### DISCUSSION THREAD

**Discussion by joaquinjsb** [baresip/baresip#2921] [Q&A] [answered]
Hello,
When I do a call from -e command, the calls starts immediately without waiting for registration.

**Comment by cspiel1** [comment]
Not sure if this is an issue. I don't think that there is a wait for registration condition for the `-e` commands.

**Comment by joaquinjsb** [comment]
yes, making the -e "/dial" not compatible / useful, so that should fire a warning or have a possibility for the command for wait for that hook/event?

**Comment by maximilianfridrich** [comment]
It sounds like you are using BareSIP in ways where simply calling it from the command line with some option is not enough. You might have to develop a small application that listens to BareSIP events and then triggers certain events. That's what the multiple BareSIP interfaces have been developed for.

**Comment by joaquinjsb** [comment]
yes, I guess I should use the stdin at this point, or try the D-Bus

**Comment by joaquinjsb** [comment]
sorry for one more, and I hope last ping / tag,

@maximilianfridrich do you guys any index / file I can use for reference for the list of "events" that the baresip fires? so I can try to catch them, for example, when registration is successful ?

**Comment by maximilianfridrich** [comment]
Yes, see https://github.com/baresip/baresip/blob/main/include/baresip.h#L825-L862.

**Comment by joaquinjsb** [comment]
I started working on it, I just found the https://github.com/baresip/baresip/blob/main/modules/ctrl_tcp/ctrl_tcp.c,
but I'm not receiving events in "real time", even sending a message, I'm getting no response from the server on port 4444, while using the 5555 standard, it's properly sending the std output of the process, is it working properly as of now?

**Reply by maximilianfridrich** [reply]
In your BareSIP global config
```
ctrl_tcp_listen		0.0.0.0:4444 # ctrl_tcp - TCP interface JSON
module_app		ctrl_tcp.so
```
and then listen with `nc localhost 4444`.

**Reply by joaquinjsb** [reply]
yes, I enabled it properly, looks like it's an issue on java linux socket:
https://stackoverflow.com/questions/31269870/netcat-communication-in-java

I'll try giving a look to it.

thanks

**Comment by gsker** [comment]
This is an unanswered discussion, so I hope it's okay to reply to it.
I just spent a lot of time trying to make a call work from the linux commandline not realizing that this was the cause.

My solution was to manually register first with a second **`-e`**:
    **`baresip -e "/uareg" -e "d XXXXXXXXXX@chicago3.voip.ms"`**

**Comment by cspiel1** [comment, marked as answer]
As already mentioned https://github.com/baresip/baresip/discussions/2921#discussioncomment-8449998 using `-e` option in order to specify a workflow like this is not possible.

At least you have to specify in account file, load the account.so module and then you could call
```
baresip -e "d XXXXXXXXXX"
```

account
```
<sip:you@chicago3.voip.ms>;regint=3600;auth_pass=***
```

See https://github.com/baresip/baresip/wiki/Accounts !
